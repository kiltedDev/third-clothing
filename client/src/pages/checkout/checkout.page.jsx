import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { selectCartItems, selectCartTotal } from '../../redux/cart/cart.selectors';
import CheckoutItem from '../../components/checkout-item/checkout-item.component';
import StripeCheckoutButton from '../../components/stripe-button/stripe-button.component';

import {
  CheckoutPageContainer,
  CheckoutHeader,
  CheckoutTotal,
  HeaderBlock,
  TestWarning
} from './checkout.styles';

export const CheckoutPage = ({ cartItems, cartTotal }) => {
  return(
    <CheckoutPageContainer className='checkout-page'>
      <CheckoutHeader className='checkout-header'>
        <HeaderBlock className='header-block'>
          <span>Product</span>
        </HeaderBlock>
        <HeaderBlock className='header-block'>
          <span>Description</span>
        </HeaderBlock>
        <HeaderBlock className='header-block'>
          <span>Quantity</span>
        </HeaderBlock>
        <HeaderBlock className='header-block'>
          <span>Price</span>
        </HeaderBlock>
        <HeaderBlock className='header-block'>
          <span>Remove</span>
        </HeaderBlock>
      </CheckoutHeader>
      {
        cartItems.map(cartItem =>
          <CheckoutItem key={ cartItem.id } cartItem={ cartItem } />
        )
      }
      <CheckoutTotal
        className='total'
        Children={ `Total: ${ cartTotal }` }
      />
    <TestWarning className='test-warning'>
        * Please use the following test credit card for payment*
        <br />
        4242 4242 4242 4242
        <br />
        Exp: 01/24 - CVV: 123
      </TestWarning>
      <StripeCheckoutButton price={ cartTotal } />
    </CheckoutPageContainer>
  )
}

const mapStateToProps = createStructuredSelector({
  cartItems: selectCartItems,
  cartTotal: selectCartTotal
})

export default connect(mapStateToProps)(CheckoutPage);
