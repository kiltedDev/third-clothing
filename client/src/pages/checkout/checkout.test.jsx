import { shallow } from 'enzyme';
import React from 'react';
import { CheckoutPage } from './checkout.page';

describe('CheckoutPage component', () => {
  let wrapper;
  let snapshot;
  beforeAll(() => {
    const mockProps = {
      cartItems: [],
      cartTotal: 144
    };
    wrapper = shallow(<CheckoutPage { ...mockProps } />);
    snapshot = wrapper.debug();
  });

  it('should render CheckoutPage component',() => {
    expect( snapshot ).toMatchSnapshot();
  });
});
