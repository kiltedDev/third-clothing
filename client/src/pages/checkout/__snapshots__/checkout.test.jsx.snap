// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutPage component should render CheckoutPage component 1`] = `
"<styled.main className=\\"checkout-page\\">
  <styled.div className=\\"checkout-header\\">
    <styled.div className=\\"header-block\\">
      <span>
        Product
      </span>
    </styled.div>
    <styled.div className=\\"header-block\\">
      <span>
        Description
      </span>
    </styled.div>
    <styled.div className=\\"header-block\\">
      <span>
        Quantity
      </span>
    </styled.div>
    <styled.div className=\\"header-block\\">
      <span>
        Price
      </span>
    </styled.div>
    <styled.div className=\\"header-block\\">
      <span>
        Remove
      </span>
    </styled.div>
  </styled.div>
  <styled.div className=\\"total\\" Children=\\"Total: 144\\" />
  <styled.div className=\\"test-warning\\">
    * Please use the following test credit card for payment*
    <br />
    4242 4242 4242 4242
    <br />
    Exp: 01/24 - CVV: 123
  </styled.div>
  <Connect(StripeCheckoutButton) price={144} />
</styled.main>"
`;
